adapters/repos/db/helpers/allow_list.go

```diff
@@ -20,6 +20,7 @@ type AllowList interface {
 	Insert(ids ...uint64)
 	Contains(id uint64) bool
 	DeepCopy() AllowList
+	WrapCopy() AllowList
```

Comment:
Nitpicking, please feel free to ignore: I feel something like `CopyOnWrite` would be more explicit